x/vbr/types/msgs_test.go

```diff
@@ -1,132 +1,172 @@
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			msg := &MsgIncrementBlockRewardsPool{
+				Funder: tt.fields.Funder,
+				Amount: tt.fields.Amount,
+			}
+			if err := msg.ValidateBasic(); (err != nil) != tt.wantErr {
+				t.Errorf("MsgIncrementBlockRewardsPool.ValidateBasic() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
 
 // -------------------------
 // --- MsgSetParams
 // -------------------------
-var TestGov, _ = sdk.AccAddressFromBech32("cosmos1lwmppctrr6ssnrmuyzu554dzf50apkfvd53jx0")
-var TestEarnRate = sdk.NewDecWithPrec(5, 1)
-
-var msgSetParams = MsgSetParams{
-	Government:           TestGov.String(),
-	DistrEpochIdentifier: EpochDay,
-	EarnRate:             TestEarnRate,
-}
-
-var msgSetParamsNoEpochIdentifier = MsgSetParams{
-	Government:           TestGov.String(),
-	DistrEpochIdentifier: "",
-	EarnRate:             TestEarnRate,
-}
 
 func TestMsgSetParams_Route(t *testing.T) {
-	actual := msgSetParams.Route()
-	expected := ModuleName
+	expected := RouterKey
+	actual := ValidMsgSetParams.Route()
 
 	require.Equal(t, expected, actual)
 }
 
 func TestMsgSetParams_Type(t *testing.T) {
-	actual := msgSetParams.Type()
 	expected := MsgTypeSetParams
+	actual := ValidMsgSetParams.Type()
 
 	require.Equal(t, expected, actual)
 }
 
-func TestMsgSetParams_valid(t *testing.T) {
-	actual := msgSetParams.ValidateBasic()
-
-	require.Nil(t, actual)
-}
-
-func TestMsgSetParams_ValidateBasic_noEpochIdentifier(t *testing.T) {
-	actual := msgSetParamsNoEpochIdentifier.ValidateBasic()
-	expected := sdkErr.Wrap(sdkErr.ErrInvalidType, fmt.Sprintf("invalid epoch identifier: %s", msgSetParamsNoEpochIdentifier.DistrEpochIdentifier))
-
-	require.Equal(t, expected.Error(), actual.Error())
-}
-
 func TestMsgSetParams_GetSignBytes(t *testing.T) {
-	actual := msgSetParams.GetSignBytes()
-	expected := sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&msgSetParams))
+	expected := sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&ValidMsgSetParams))
+	actual := ValidMsgSetParams.GetSignBytes()
 
 	require.Equal(t, expected, actual)
 }
 
 func TestMsgSetParams_GetSigners(t *testing.T) {
-	actual := msgSetParams.GetSigners()
-	govAddr, _ := sdk.AccAddressFromBech32(msgSetParams.Government)
-	expected := []sdk.AccAddress{govAddr}
+	expectedAddr, _ := sdk.AccAddressFromBech32(ValidMsgSetParams.Government)
+	expected := []sdk.AccAddress{expectedAddr}
+	actual := ValidMsgSetParams.GetSigners()
 
 	require.Equal(t, expected, actual)
 }
+
+func TestMsgSetParams_ValidateBasic(t *testing.T) {
+	type fields struct {
+		Government           string
+		DistrEpochIdentifier string
+		EarnRate             sdk.Dec
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		wantErr bool
+	}{
+		{
+			name:    "ok",
+			fields:  fields(ValidMsgSetParams),
+			wantErr: false,
+		},
+		{
+			name: "invalid Government",
+			fields: fields{
+				Government:           "",
```

Comment:
This return an error because this field is not a valid account address, and not because the account isn't the government. It should be better change test name to "invalid address" or  "invalid Government address format" (and maybe should be better change the output of the error)